sdk/src/event_listener/tangle/mod.rs

```diff
@@ -78,10 +75,32 @@ impl<Ctx, Evt> TangleEvent<Ctx, Evt> {
     }
 }
 
-impl<Ctx> IsTangle for TangleEventListener<Ctx> {}
+pub trait EventMatcher: Send + 'static {
+    type Output: Send + 'static;
+    fn try_decode(event: EventDetails<TangleConfig>) -> Option<Self::Output>;
+}
+
+impl<T: StaticEvent + Send + 'static> EventMatcher for T {
+    type Output = T;
+    fn try_decode(event: EventDetails<TangleConfig>) -> Option<Self::Output> {
+        event.as_event::<T>().ok().flatten()
+    }
+}
+
+#[derive(Copy, Clone)]
+pub struct AllEvents;
+
+impl EventMatcher for AllEvents {
+    type Output = EventDetails<TangleConfig>;
+    fn try_decode(event: EventDetails<TangleConfig>) -> Option<Self::Output> {
+        Some(event)
+    }
+}
+
+impl<Ctx, Evt: EventMatcher> IsTangle for TangleEventListener<Ctx, Evt> {}
 
 #[async_trait]
-impl<Ctx: Clone + Send + Sync + 'static, Evt: Send + Sync + 'static>
+impl<Ctx: Clone + Send + Sync + 'static, Evt: EventMatcher>
```

Comment:
```suggestion
impl<C: Clone + Send + Sync + 'static, E: EventMatcher>
```
It would be good to also create a trait that wrap `Clone + Send + Sync + 'static`